docs/modules/sql/pages/user-defined-types.adoc

```diff
@@ -64,30 +51,34 @@ To create a new type, use the `CREATE TYPE` statement:
 
 [source,sql]
 ----
-CREATE [OR REPLACE] TYPE [IF NOT EXISTS] MyTypeName
-    [(colName colType, ...)]
-    OPTIONS (
-    'format'='{java|portable|compact}'
-    [, 'javaClass'='com.myPackage.MyJavaClass'] <1>
-    [, 'compactTypeName'='MyCompactRecordTypeName'] <2>
-    [, 'portableFactoryId'='123', 'portableClassId'='456', ['portableVersion'='789']] <3>
-    )
+CREATE [OR REPLACE] TYPE [IF NOT EXISTS] MyTypeName [(
+    colName colType,
+    ...
+)] [OPTIONS (
+    'javaClass'='com.mypackage.MyJavaClass' <1>
+  | 'compactTypeName'='MyCompactRecordName' <2>
+  | 'portableFactoryId'='123', 'portableClassId'='456', ['portableVersion'='789'] <3>
+  | 'avroSchema'='{"type":"record","name":"myType","fields":[{"name":"colName","type":"colType"},...]}' <4>
+)]
 ----
-<1> `java` format requires the `javaClass` option.
-<2> `portable` format requires `portableFactoryId` and `portableClassId` and optionally `portableVersion`.
-<3> `compact` format requires `compactTypeName` - this is not the name of the created type, but rather internal name of the Compact record type, used internally by the Compact Serialization format.
+<1> In `java` format, you can use `javaClass` option to override the type class with a subclass. Note that you must provide a concrete class for `INSERT` statement to work, which makes `javaClass` option mandatory when the original type class is abstract or interface.
```

Comment:
"You must provide a concrete class for `INSERT` statements" can be replaced with "You must provide a concrete class to be able to `INSERT`".

Reply:
No, it cannot. "to be able to" is not a good phrase.

How about "You must provide a concrete class to use `INSERT` statements" ?

Reply:
But then the sentence has no agreement: use INSERT statements vs use an INSERT statement

statements is the correct term here.

Reply:
**The INSERT statement** is a type of SQL query
**An INSERT statement** is an instance of the INSERT SQL query type
**INSERT statements** is a generalised way of referring to INSERT SQL queries

I may not be a programmer, but I have worked in software development for over 30 years.

I really appreciate the work you are putting into this, and the time you are taking to explain things (such as the tuples conversation). However, on this point, I remain unmovable.

How about: "You must provide a concrete class for each `INSERT` statement;", would that allow us to agree?

Reply:
This is not really about software development. In English, a noun can refer to a "concept" instead of a particular instance (or instances). In such a case, it becomes uncountable, and consequently singular. For example, we can say that "police closed the investigation" (without a/the/-s) if we are not interested in a particular police. You must have seen such usages before in the news(papers). "`INSERT` statement" is not wrong (even without "the"), and it looks more natural to me. If you disagree, your suggestion can stay as it is.

Reply:
I disagree.

I understand, but, in this case `INSERT` = police, `INSERT` statement = the police. 

Police is a group term, or with your obvious mathematical bias, uncountable. Police is used when speaking of the force in general.
The police is specific, hence the definite article, used when we think the reader knows what we are referring to. 
The plural of police is still police.
You would never say a police; it would be a policeman, an officer, a policewoman.

This isn't a news(paper). It is technical communication. We need to be clear (hence why we don't use LaTex and mathematical equations, which "might be a bit confusing to the reader"). We need to make sure our information is clear and usable, with no misinterpretation, or cognitive dissonance.

You can check Microsoft's SQL docs to check the use if you like; for example: https://learn.microsoft.com/en-us/sql/t-sql/queries/top-transact-sql?view=sql-server-ver16